File: php/draft_comments.php
<?php

require_once 'entity.php';
require_once 'drafts.php';
require_once 'users.php';
require_once 'files.php';
require_once 'notifications.php';

class DraftComment extends Entity
{
    public static $resource = '/draft_comments/';
    public static $json_name = 'draft_comment';

    public function __construct()
    {
        $this->json_property('id','integer');
        $this->json_property('draft','Draft', $many = False,
                              $recursive = True);
        $this->json_property('job', 'Job', $many = False,
                              $recursive = True);
        $this->json_property('user', 'User', $many = False,
                              $recursive = True);
        $this->json_property('fowards', 'User', $many = True,
                              $recursive = True);
        $this->json_property('notifications', 'Notification', $many = True,
                              $recursive = True);
        $this->json_property('urgency','integer');
        $this->json_property('subject','string');
        $this->json_property('date', 'DateTime');
        $this->json_property('text', 'string');
        $this->json_property('change_request', 'boolean');
        $this->json_property('send_sms', 'boolean');
        $this->json_property('send_email', 'boolean');
    }
}

class DraftComments extends Resource
{
    public function __construct()
    {
        $this->entity_class = 'DraftComment';
        $this->json_name = 'draft_comments';
    }
}

$draft_comments = new DraftComments();

File: python/draft_comments.py
import datetime
import sdk.python.entities
from sdk.python.entities import Property
from sdk.python.files import File


class DraftComment(sdk.python.entities.Entity):

    resource = '/draft_comments/'
    json_name = 'draft_comment'

    id = Property(int)
    files = Property(File)
    urgency = Property(int)
    subject = Property(str)
    date = Property(datetime.datetime)
    text = Property(str)
    change_request = Property(bool)
    send_sms = Property(bool)
    send_email = Property(bool)


class DraftComments(sdk.python.entities.Resource):

    entity_class = DraftComment
    json_name = 'draft_comments'


draft_comments = DraftComments()
